src/handler/events/readyHandler.ts

```diff
@@ -0,0 +1,160 @@
+import { EventsHandler } from './eventsHandler';
+import type Wrapper from '../structures/wrapper';
+import { concatMap, fromEvent, Observable, map, take, of, from, toArray, switchMap } from 'rxjs';
+import type { CommandModule } from '../structures/module';
+import * as Files from '../utilities/readFile';
+import { errTap } from './observableHandling';
+import type { DefinedCommandModule } from '../../types/handler';
+import { basename } from 'path';
+import { CommandType, PayloadType, PluginType } from '../structures/enums';
+import { processCommandPlugins } from './userDefinedEventsHandling';
+import type { Awaitable } from 'discord.js';
+import type { Result } from 'ts-results';
+import { SernError } from '../structures/errors';
+import { match } from 'ts-pattern';
+import { Err, Ok } from 'ts-results';
+import { ApplicationCommandType, ComponentType } from 'discord.js';
+
+export default class ReadyHandler extends EventsHandler<{
+    mod: DefinedCommandModule;
+    absPath: string;
+}> {
+    protected discordEvent!: Observable<{ mod: CommandModule; absPath: string }>;
+    constructor(wrapper: Wrapper) {
+        super(wrapper);
+        const ready$ = fromEvent(this.wrapper.client, 'ready').pipe(take(1));
+        this.discordEvent = ready$.pipe(
+            concatMap(() =>
+                Files.buildData<CommandModule>(this.wrapper.commands).pipe(
+                    errTap(reason =>
+                        wrapper.sernEmitter?.emit('module.register', {
+                            type: PayloadType.Failure,
+                            module: undefined,
+                            reason,
+                        }),
+                    ),
+                ),
+            ),
+        );
+        this.init();
+        this.payloadSubject
+            .pipe(
+                concatMap(payload => this.processPlugins(payload)),
+                concatMap(payload => this.resolvePlugins(payload)),
+            )
+            .subscribe(payload => {
+                const allPluginsSuccessful = payload.pluginRes.every(({ execute }) => execute.ok);
+                if (allPluginsSuccessful) {
+                    const res = registerModule(payload.mod);
+                    if (res.err) {
+                        throw Error(SernError.NonValidModuleType);
+                    }
+                    wrapper.sernEmitter?.emit('module.register', {
```

Comment:
how bout building another enum for all sern events?
enums are style of v14 so 👀

Reply:
we will see. You can register events by file name, so not really sure how enums would play into this

Reply:
```ts
enum Events {
	ModuleRegister = 'module.register'
	// etc
}
```

Reply:
also wait why would file name be responsible for events
i don't need to have the event name same as file name, so i should be asked for an event name
now we can combine djs event names and sern events name and export it as `Events` so users can use it